### DISCUSSION TITLE
Suggestion for Enhancing the "Quick Open" Menu Experience in Godot [w/ Prototype]

### DISCUSSION THREAD

**Discussion by kevincardona** [godotengine/godot-proposals#7717] [Editor]
I feel like the quick open menu should feel a little bit more seamless to the editor. My bigest gripe with the pre-existing implementation is that you cant just deblur/defocus to close the menu. It also think that having a separated native window makes it feel a bit more disconnected from the editor.

I think I would value having quick open be attached to the editor (think VS Code file search) and maybe using a PopupPanel (or new subclass) over the ConfirmationDialog and I wanted to get everyones thoughts on that.

I made a prototype that makes these changes that you can take a look at [here](https://github.com/kevincardona/godot/commit/cac84b5d64cf72e15e5aa8f80652f5529883ac04). It also includes 
an additional "No results for ..." dialog that I think would be nice. 

Note: The styling isn't perfect in my prototype

https://github.com/godotengine/godot-proposals/assets/11409047/9417ea88-a616-45af-b937-3e886713cc8e

**Comment by Calinou** [comment]
This was being worked on at some point by another contributor: https://github.com/godotengine/godot/pull/56772

That PR fell behind and has a lot of merge conflicts now, so it would probably need to be redone. I liked its design but it would also need some tweaks to better match with the rest of the Godot editor.

**Reply by kevincardona** [reply]
I totally missed that! I'll take a look at that
